### ISSUE TITLE
Handling of multiple texts / versions of a text

### ISSUE THREAD

**Issue by RobertoRDT** [evt-project/evt-viewer#10]
It has been a long-standing feature request, and there have been several user inquiries in this sense, that EVT support display and comparison of different versions of a text, i.e. main/base text, one or more translations, a version in another language, a subsequent reworked version of the same text, a possible source etc.

_Request by: Dominkika Heublein
Reference: 2018-01-11 mail to evt.developers_

> I am a research assistant at University of Würzburg, Germany, and right now working for „Narragonien digital“ (Link: http://kallimachos.de/kallimachos/index.php/Narragonien:Main ). We are currently creating TEI-documents out of OCR-Texts we got from the original manuscripts. Our next future step would be to present our edition on the web. While doing some research on that topic I found the tool you created and tried some basic things with the download version, which means, I haven’t cloned the github repository yet. Now my actual question: We would like to create a synoptic viewer which allows the user to choose which text or facsimile they want to compare to which text or facsimile. In fact, we have around 15 sub-editions inside of a big edition and we want to be able to compare specific chapters of different sub-editions next to each other. Do you think it is possible to integrate such a functionality into your tool? I would be very glad to hear about your thoughts on that.

> I worked myself into your EVT2 Demo and it seems that the multiple recension feature could be adapted for our case. I think the main difference with our texts is the fact that they are not at all the same – semantically, yes. They all derive from one source document, which is a German text. Afterwards other German texts developed, but also English, French, Latin and Flemish translations, which do not translate word for word. A comparison between the texts rather makes sense on chapter basis. The documents are separate – each “book” has its own TEI document; within each TEI document furthermore there is a normalized and an OCR version of the text.

> All in all: Many translations of one text, which could be made with the “Recension” Part of your developer, right? Within one translation there is a “normalized” and “OCR” Version – so this would go with the “Version-Tab”? Faksimile also should be included.

_Request by: Hunaynnet project [Nicolas Bamballi, Slavomir Ceplo]
Reference: 2018-03-09, 2018-04-09 mail to evt.developers_

> [Nicolas]We are currently preparing a digital parallel corpus of Greek philosophical and scientific texts (of e.g. Aristotle, Galen, et al.) and their translations into Syriac and Arabic. At the moment we have plain transcriptions of all the texts (about Greek 15 works and Syriac and Arabic 60 versions) and are exploring the possibilities for digital presentation. The way will proceed is to prepare the texts in all three languages with Classical Text Editor and then export them into XML for the purpose of the digital presentation.

> As far as I see [...] a layout such as the one you currently have for the Avicenna text in EVT 2 would perfectly suit our needs. The one obvious change would be that instead of having a presentation with parallel texts as *witnesses* we would rather have a source text (Greek) and *versions* (Syriac and Arabic). Accordingly, our intention is to be able to display the alignment units across versions I have just mentioned in the same way in which variants are displayed across witnesses in the Avicenna sample.
In addition to the structure Nicolas described, we intend to implement some additional functionality:

> [Slavomir] We've aligned the text by sentence (-ish) in all versions by using CTE synchronization which translates to the <anchor type="sync2" /> TEI XML element. We would like to provide an option to visualize this synchronization, say, by hovering over some synchronization mark whereupon the equivalent sentence-like units would be highlighted in all displayed versions. See the current beta version at http://bulbul.sk/HUNAYNNET/de_interpretatione.html for what we have in mind where this is achieved by a combo of XSLT and JQuery (please don't be too harsh, this is all very preliminary, more omega than beta :).

_Request by: Pietro Liuzzo
Reference: 2018-09-29, 2018-04-09 mail to evt.developers_

> Nel contesto di un breve survey sulle modalità di visualizzare edizioni critiche da TEI, vorrei visualizzare con EVT-2 un breve testo di cui, nel medesimo file TEI ho una edizione del testo Etiopico, una del testo Latino, entrambe con il proprio apparato inline e una traduzione.
come è meglio procedere? forse devo dividere e tenere ciascuno dei testi in un file separato? come devo impostare config.json per poter vedere i quattro testi uno accanto all'altro?

**Comment by RobertoRDT** [comment]
A first hypothesis of implementation:

- multiple texts are showed using an instance of the **collation view** (base text +automatically generated witnesses) _and_ the **text+text** view;
- this view will be renamed so to make it clear that it's "base text + alternative versions" of a text (therefore not an actual collation view), provided with new icon and all;
- each text, be it the main one or other versions of the same, will be marked with @xml:id and, for clarity, @type as well (for instance: type="base_text", type="it_trans", type="lat_version", type="greek_source" etc.), @n optional as usual;
- in the configuration options there will be BASE TEXT = [corresponding xml:id value], ALT VERSION 1 = [corresponding xml:id value], ALT VERSION 2 = [corresponding xml:id value] etc.;
- supplementary option to determine what is going to be loaded automatically at the start: LOAD TIME = BASE TEXT; ALT VERSION 1 etc.; all texts, however, will be available by means of the drop down menu and/or the + button (just as for the automatically generated witnesses in the collation view);
- at the moment of parsing and load time each text will be loaded in its frame and displayed as specified in the configuration file;
- in the text-text view, on the other hand, the user will always see the BASE TEXT and the first corresponding ALT version, again as per user configuration; it will however be possible to choose what to display by means of the drop-down menus, the advantage with respect to the collation view being that for each text 50% of the screen will be available.

This way the demand on the user part is negligible (just adding @xml:id and a @type to each text) and the changes to implement are minimal.

**Comment by ChiaraMartignano** [comment]
I agree on @RobertoRDT's last comment, especially for the visualization part. But I have an alternative solution on how to define the texts in the configuration file, that should make **setting the configuration** file itself much easier.
In order to parse the file correctly, EVT needs to know only which is the main document of the file. This piece of information can be specified by assigning the id (i.e. value of xml:id) of the main document to the property named "**mainDocId**" in the config file.
If the other documents in the file correspond to the texts of the witnesses, in the `<listWit>`, each `<witness>` can be bound to the corresponding document through the `@corresp` attribute carrying the id of the document.
If the other documents aren't witnesses, they will be displayed in the text-text view mode as mentioned in the comment above.

**Comment by RobertoRDT** [comment]
Actually the point of this feature is that the relationship between the main or "privileged" version of a text is not the same as that in a "critical text" <-> "all related witnesses one". Which is also why I wouldn't want to mix this view with the standard collation view, it has to be clearly separate, then if you have a critical text and want to explore its variance you can open the collation view. Think about the "original text" plus "translated texts" need: you may want to compare the first and one or more of the latter, but perhaps you may also be interested in comparing translations only, which would be unlikely in a "base text - witnesses" collation such as the current one. It may also happen the case when you consider all the different versions equally important: see the Hunaynnet project needs.

I would think that we could take advantage of the collation view for this purpose, besides the text-text (only two texts at the same time) view, but the latter with its "agnosticism" with regard to what is the "main" text should be the natural place where to load texts. If you think that it is a distraction to have two separate views to do the same thing (I'm thinking this is a sensible objection to my initial proposal) we could modify the collation view to work like the text-text view only allowing to compare more than two texts.

I guess we also need to do a little more UI testing via mockups.

**Comment by ctupman** [comment]
Are there any plans to implement support for translations during this coming year's work on EVT? This is something that would be of particular interest to one of our current projects, in which we want to display both a transcription and a translation.

**Comment by RobertoRDT** [comment]
> Are there any plans to implement support for translations during this coming year's work on EVT? This is something that would be of particular interest to one of our current projects, in which we want to display both a transcription and a translation.

We are going to release EVT2js beta 2 next spring, but I am not sure that this feature will make it into this version. The dev team is now working on porting the code base to Angular 8, which may well mean that this feature will be added after the first release based on the new framework will be out. The good news is that this is a very important feature, requested by many scholars, so it will be definitely implemented.
